WeatherApp/Services/ParsingService.swift

```diff
@@ -12,30 +12,28 @@ class ParsingService {
     
     private let decoder = JSONDecoder()
     
-    func parseCityWeather(_ data: Data, city: String) -> CityWeather? {
-        guard let currentWeatherResponse = try? decoder.decode(WeatherResponse.self, from: data) else {
+    func parseCurrentWeather(_ data: Data) -> CurrentWeatherResponse? {
+        guard let currentWeatherResponse = try? decoder.decode(CurrentWeatherResponse.self, from: data) else {
```

Comment:
Why don't you just return `try? decoder.decode(CurrentWeatherResponse.self, from: data)` since the return type is optional. Please change this at all places in this class where applicable.